iotfunctions/anomaly.py

```diff
@@ -431,93 +519,218 @@ def __init__(self, input_item, windowsize, output_item):
+            #   for missing data detection we look at the timestamp gradient instead
+            dfe, temperature = self.prepare_data(dfe)
 
-            #  interpolate gaps - data imputation
-            # Size = dfe[[self.input_item]].fillna(0).to_numpy().size
-            dfe = dfe.interpolate(method='time')
-
-            # one dimensional time series - named temperature for catchyness
-            temperature = dfe[[self.input_item]].fillna(0).to_numpy().reshape(-1,)
-
-            logger.debug('KMeans: ' + str(entity) + ', ' + str(self.input_item) + ', ' + str(self.windowsize) + ', ' +
-                         str(self.output_item) + ', ' + str(self.step) + ', ' + str(temperature.size))
+            logger.debug('Module GeneralizedAnomaly, Entity: ' + str(entity) + ', Input: ' + str(self.input_item) +
+                         ', Windowsize: ' + str(self.windowsize) + ', Output: ' + str(self.output_item) +
+                         ', Overlap: ' + str(self.step) + ', Inputsize: ' + str(temperature.size))
 
             if temperature.size > self.windowsize:
-                logger.debug(str(temperature.size) + ',' + str(self.windowsize))
+                logger.debug(str(temperature.size) + "," + str(self.windowsize))
 
-                # Chop into overlapping windows
-                slices = skiutil.view_as_windows(temperature, window_shape=(self.windowsize,), step=self.step)
+                # NN = GeneralizedAnomalyModel( base_learner=MinCovDet(), fit_function="fit",
+                #        predict_function="mahalanobis", score_sign=1,)
+                temperature -= np.mean(temperature, axis=0)
+                mcd = MinCovDet()
 
-                if self.windowsize > 1:
-                    n_clus = 40
-                else:
-                    n_clus = 20
+                # Chop into overlapping windows (default) or run through FFT first
+                slices = self.feature_extract(temperature)
+
+                pred_score = None
 
-                cblofwin = CBLOF(n_clusters=n_clus, n_jobs=-1)
                 try:
-                    cblofwin.fit(slices)
+                    # NN.fit(slices)
+                    mcd.fit(slices)
+
+                    # pred_score = NN.decision_function(slices).copy()
+                    pred_score = mcd.mahalanobis(slices).copy()
+
+                except ValueError as ve:
+
+                    logger.info(
+                        self.whoami + " GeneralizedAnomalyScore: Entity: "
+                        + str(entity) + ", Input: " + str(self.input_item) + ", WindowSize: "
+                        + str(self.windowsize) + ", Output: " + str(self.output_item) + ", Step: "
+                        + str(self.step) + ", InputSize: " + str(slices.shape)
+                        + " failed in the fitting step with \"" + str(ve) + "\" - scoring zero")
+
+                    pred_score = np.zeros(slices.shape[0])
+                    pass
```

Comment:
Just wanted to make sure. You're printing out the stack trace and also passing the exception.

Reply:
yes  - idea is to show the exception content in the log message